File: grimdb/sdk/ruby/lib/grimlocker/audit.rb
# frozen_string_literal: true
module Grimlocker
  AuditEvent = Struct.new(:timestamp, :level, :module, :message, :subject_id) do
    def self.from_hash(h) = new(h['timestamp']||0, h['level']||'', h['module']||'', h['message']||'', h['subject_id'])
  end
end

File: grimdb/sdk/rust/src/audit.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Deserialize, Serialize)]
pub struct AuditEvent {
    pub timestamp:  i64,
    pub level:      String,
    pub module:     String,
    pub message:    String,
    pub subject_id: Option<String>,
}
